### ISSUE TITLE
Image pull secret not added for setup credentials job

### ISSUE THREAD

**Issue by johnny-gossick** [yugabyte/charts#190]
The setup credentials job is missing the image pool secret set [here](https://github.com/yugabyte/charts/blob/2d625c9cbf68ad02105706d37a372cacbe22cee4/stable/yugabyte/values.yaml#L13).  

If the user sets a custom image repository [here](https://github.com/yugabyte/charts/blob/2d625c9cbf68ad02105706d37a372cacbe22cee4/stable/yugabyte/values.yaml#L10C3-L10C13) and their registry requires authentication, the setup credentials job will throw an image ImagePullBackOff and not be able to start.

**Comment by dbblackdiamond** [comment]
I am running into the same issue. The reason for this is that the template for the setup credentials job doesn't include any key inheriting its value from the `values.yaml` file.  It could be fixed by adding a `pullSecretName` to `spec.template.spec` with a value of `{{ .Values.Image.pullSecretName }}`. This way, the `pullSecretName` defined in the `values.yaml` file get passed all the way to the setup credential job.

I can create a PR for this if there is consensus that it is the best way forward.

**Comment by amannijhawan** [comment]
@dbblackdiamond  Makes sense, if you are still interested in sending a PR. We will review it.
